### ISSUE TITLE
Deno 2.x is not available in the Arch repositories

### ISSUE THREAD

**Issue by dhtseany** [ninjadev64/OpenDeck#28]
Hello! When we try to build `opendeck` on Arch using the AUR package we're catching failures because `tauri.conf.json` was built using version 1.6.x standards but the Arch version of tauri-cli is now at 2.0.3. When we use 2.0.3 to build we're catching the following errors:

```
Initialize @esbuild/linux-x64@0.21.5
warning: Packages contained npm lifecycle scripts (preinstall/install/postinstall) that were not executed.
    This may cause the packages to not work correctly. To run them, use the `--allow-scripts` flag with `deno cache`
    (e.g. `deno cache --allow-scripts=pkg1,pkg2 <entrypoint>`):
      npm:@sveltejs/kit@2.5.26, npm:esbuild@0.21.5, npm:svelte-preprocess@5.1.4
==> Removing existing $pkgdir/ directory...
==> Starting build()...
    Error `tauri.conf.json` error: "identifier" is a required property
    Error `tauri.conf.json` error on `build`: Additional properties are not allowed ('devPath', 'distDir' were unexpected)
    Error `tauri.conf.json` error: Additional properties are not allowed ('package', 'tauri' were unexpected)
==> ERROR: A failure occurred in build().
    Aborting...
```

Our work-around is to downgrade tauri-cli to version 1.6.2 as outlined in [my AUR comment](https://aur.archlinux.org/packages/opendeck#comment-995467) then build the opendeck package. 

I'll be happy to provide any assistance I can for testing. Thanks!

**Comment by ninjadev64** [comment]
In fact, the version of OpenDeck on the AUR is out of date, OpenDeck is updated to use V2. The maintainer of OpenDeck on the AUR is slow to respond unfortunately.

**Comment by ninjadev64** [comment]
You can use the opendeck-git package in the meantime

**Comment by ninjadev64** [comment]
OpenDeck on Arch is currently in limbo, the git package won't work until Deno is updated in the Arch repositories, and the non-git package won't work as it's too old. You just have to wait for the AUR maintainer to update the non-git package to 2.1.0, which is not the latest but also not the newest (it uses the new tauri-cli which Arch has but not the new Deno which Arch doesn't).

**Comment by ninjadev64** [comment]
Hey @dhtseany 👋 

I see you've adopted the deno-git package on the AUR to get OpenDeck working, awesome!

Here are the modifications you'll need to make to the OpenDeck PKGBUILDs once deno-git builds Deno 2.0 sources correctly:

```sh
# ...
makedepends=(git deno-git dpkg hidapi webkit2gtk libappindicator-gtk3)
# ...

prepare() {
    # Rust stuff...
    export DENO_DIR="${srcdir}/deno-cache"
    deno install
}

build() {
    # Rust stuff...
    export DENO_DIR="${srcdir}/deno-cache"
    deno task tauri build --ci -b deb -- --frozen
}

# ...
```

**Comment by dhtseany** [comment]
Thanks, I'm trying to anyways! I maintain other packages on Arch and I've been searching for a forward replacement for streamdeck-ui-gui. The current maintainer wants everyone to go to the StreamController app but I'm not a fan of pip-only apps so I've been seeking out other options and came across OpenDeck. Once I saw we needed a deno package at least v2.0 I figured I could take crack at it and see if I could get it working.

At the moment I'm stuck on figuring out why the PKGBUILD produces build errors while running the same _cargo build_ command works fine if I run it "by hand". 

I'm thinking it's an external dependency but I'm still trying to learn Deno's expected build methods on the fly as I've never really worked with it before.

**Comment by ninjadev64** [comment]
Well, we're not actually building OpenDeck using Deno. We're simply using Deno as a means to install and run the Tauri CLI, which first calls upon Vite to package the frontend and then Cargo to build the Rust backend.

Previously OpenDeck used the Tauri CLI through Cargo instead of Deno, and Cargo only allows user-wide installs to a user-wide location, which meant we needed a separate tauri-cli Arch package to obtain it in order to keep the user's system non-conflicting with other applications that used the Cargo CLI.

This worked initially, then the tauri-cli package fell out of date, then it worked again for a while, then it fell out of date again, and then that worked again but OpenDeck updated to Deno v2 which wasn't available in the Arch repositories last time I checked.

Now, we're just waiting for Deno v2 to be available on Arch so you can apply the changes to the PKGBUILD I sent above to resolve all these problems, using a locally held Tauri CLI instead of the global Cargo version allowing OpenDeck to use the correct version for itself.

**Comment by ninjadev64** [comment]
By the way, as for downgrading the Tauri CLI to 1.6 as you mentioned on the AUR, you should instead update OpenDeck, as it has utilised the new Tauri version for a while now but the version of it tagged on the AUR is out of date.

Don't go too new though, versions of OpenDeck newer than 2.1.1 (excluding) use Deno v2 which isn't tagged in the Arch repositories yet. Your adopting of the deno-git package and updating it will help resolve this issue.

**Comment by ninjadev64** [comment]
It seems that Deno 2.x has been tagged in the Arch repositories. I will inform the maintainer of the OpenDeck AUR packages.

**Comment by kamack38** [comment]
It seems that the current version of the `deno` package is `2.1.2-1.1`, so the AUR package can be moved to it. Am I correct?

**Comment by ninjadev64** [comment]
Yes, the comment before yours says that. The opendeck-git package has already been updated because it has another maintainer, but that maintainer is not present on the opendeck package.

**Comment by kamack38** [comment]
I've written an email to become the maintainer of the `opendeck` package. I will also submit a request to the AUR to orphan the package, so I'll be able to adopt it.

**Comment by ninjadev64** [comment]
This is all good now, @Terrorwolf has kindly stepped up to maintain the package.
